gl_generator/generators/global_gen.rs

```diff
@@ -1,309 +1,424 @@
-fn write_type_aliases<W>(registry: &Registry, dest: &mut W) -> io::Result<()>
+/// The function calls the corresponding function pointer stored in the
+/// `storage` module created  by `write_ptrs`.
+fn write_fns<W>(
+    registry: &Registry,
+    dest: &mut W,
+    trace: bool,
+    debug_assert: bool,
+    use_atomics: bool,
+) -> io::Result<()>
 where
     W: io::Write,
 {
-    try!(writeln!(
-        dest,
-        r#"
-        pub mod types {{
-            #![allow(non_camel_case_types, non_snake_case, dead_code, missing_copy_implementations)]
-    "#
-    ));
-
-    try!(super::gen_types(registry.api, dest));
-
     writeln!(
         dest,
-        "
-        }}
-    "
-    )
-}
-
-/// Creates all the `<enum>` elements at the root of the bindings.
-fn write_enums<W>(registry: &Registry, dest: &mut W) -> io::Result<()>
-where
-    W: io::Write,
-{
-    for enm in &registry.enums {
-        try!(super::gen_enum_item(enm, "types::", dest));
-    }
-
-    Ok(())
-}
+        r#"pub use functions::*;
+    pub mod functions {{
+      use super::*;"#
+    )?;
 
-/// Creates the functions corresponding to the GL commands.
-///
-/// The function calls the corresponding function pointer stored in the `storage` module created
-///  by `write_ptrs`.
-fn write_fns<W>(registry: &Registry, dest: &mut W) -> io::Result<()>
-where
-    W: io::Write,
-{
-    for cmd in &registry.cmds {
-        if let Some(v) = registry.aliases.get(&cmd.proto.ident) {
-            try!(writeln!(dest, "/// Fallbacks: {}", v.join(", ")));
+    for cmd in registry.cmds() {
+        // generate rustdoc link to the Kronos docs.
+        match registry.api() {
+            Api::Gl => {
+                writeln!(dest, "/// See [gl{name}](https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/gl{name}.xhtml)",
+          name = cmd.proto.ident,
+        )?;
+            },
+            Api::Gles2 => {
+                if registry.version() == (2, 0) {
+                    writeln!(dest, "/// See [gl{name}](https://www.khronos.org/registry/OpenGL-Refpages/es2.0/html/gl{name}.xhtml)",
+            name = cmd.proto.ident,
+          )?;
+                } else if registry.version() == (3, 0) {
+                    writeln!(dest, "/// See [gl{name}](https://www.khronos.org/registry/OpenGL-Refpages/es3.0/html/gl{name}.xhtml)",
+            name = cmd.proto.ident,
+          )?;
+                } else if registry.version() == (3, 1) {
+                    writeln!(dest, "/// See [gl{name}](https://www.khronos.org/registry/OpenGL-Refpages/es3.1/html/gl{name}.xhtml)",
+            name = cmd.proto.ident,
+          )?;
+                } else if registry.version() == (3, 2) {
+                    writeln!(dest, "/// See [gl{name}](https://www.khronos.org/registry/OpenGL-Refpages/es3/html/gl{name}.xhtml)",
+            name = cmd.proto.ident,
+          )?;
+                }
+            },
+            // TODO: provide docs links for more types of API.
+            _ => (),
+        }
```

Comment:
This is a super nifty/nice/kind idea! I wonder if we should move this to the  registry module somehow? That way other generators can use it.

Reply:
moved up into the registry